packages/gradle-plugin/src/main/kotlin/io/realm/gradle/RealmAnalytics.kt

```diff
@@ -0,0 +1,276 @@
+// long long time. Since Realm is a free product without an email signup, we
+// feel this is a necessary step so we can collect relevant data to build a
+// better product for you.
+//
+// Currently the following information is reported:
+// - What version of Realm is being used
+// - What OS you are running on
+// - An anonymized MAC address and bundle ID to aggregate the other information on.
+
+private const val CONNECT_TIMEOUT = 4000L
+private const val READ_TIMEOUT = 2000L
+private const val TOKEN = "ce0fac19508f6c8f20066d345d360fd0"
+private const val EVENT_NAME = "Run"
+private const val URL_PREFIX = "https://webhooks.mongodb-realm.com/api/client/v2.0/app/realmsdkmetrics-zmhtm/service/metric_webhook/incoming_webhook/metric?data="
+
+internal class RealmAnalytics : TaskExecutionAdapter() {
+    companion object {
+        @Volatile
+        var METRIC_PROCESSED = false // prevent duplicate reports being sent from the same build run
+    }
+
+    override fun afterExecute(task: Task, state: TaskState) {
+        if (!state.skipped && task.name.startsWith("compile")) {
+            sendMetricIfNeeded(task.project)
+        }
+    }
+
+    private fun sendMetricIfNeeded(project: Project) {
+        if (!METRIC_PROCESSED) {
+            val disableAnalytics: Boolean = project.gradle.startParameter.isOffline || "true".equals(System.getenv()["REALM_DISABLE_ANALYTICS"], ignoreCase = true)
+            if (!disableAnalytics) {
+                val json = jsonPayload(project)
+                sendAnalytics(json)
+            }
+            METRIC_PROCESSED = true
+        }
+    }
+
+    private fun jsonPayload(project: Project): String {
+        val userId: String = ComputerIdentifierGenerator.get()
+        val appId: String = anonymousAppId(project)
+        val osType = System.getProperty("os.name")
+        val osVersion = System.getProperty("os.version")
+
+        return """{
+                   "event": "$EVENT_NAME",
+                   "properties": {
+                      "token": "$TOKEN",
+                      "distinct_id": "$userId",
+                      "Anonymized MAC Address": "$userId",
+                      "Anonymized Bundle ID": "$appId",
+                      "Binding": "kotlin",
+                      "Realm Version": "${RealmCompilerSubplugin.version}",
+                      "Host OS Type": "$osType",
+                      "Host OS Version": "$osVersion",
+                   }
+                }"""
+    }
+
+    private fun anonymousAppId(project: Project): String {
+        var projectName = project.rootProject.name
+        if (projectName.isEmpty()) {
+            projectName = project.name
+        }
+
+        var packageName = project.group.toString()
+        if (packageName.isEmpty()) {
+            packageName = project.rootProject.group.toString()
+        }
+
+        return hexStringify(sha256Hash("$packageName.$projectName".toByteArray()))
```

Comment:
If this works, it seems like a nicer way to track builders 👍 . The transformer approach of hashing model classes felt a bit weird.